driver-core/src/main/com/mongodb/AutoEncryptionSettings.java

```diff
@@ -392,6 +436,39 @@ public boolean isBypassAutoEncryption() {
         return bypassAutoEncryption;
     }
 
+    /**
+     * Gets the mapping of a collection namespace to encryptedFields.
+     *
+     * <p><strong>Note:</strong> only applies to FLE 2. Automatic encryption in FLE 2 is configured with the encryptedFields.
+     * <p>If a collection is present in both the {@code encryptedFieldsMap} and {@link #schemaMap}, libmongocrypt will error on
+     * initialization.
+     * <p>If a collection is present on the {@code encryptedFieldsMap}, the behavior of {@code createCollection()} and
+     * {@code collection.drop()} is altered.
+     *
+     * <p>If a collection is not present on the {@code encryptedFieldsMap} a server-side collection {@code encryptedFieldsMap} may be
+     * used by libmongocrypt.
+     *
+     * @return the mapping of the collection namespaces to encryptedFields
+     * @since 4.7
+     */
+    @Nullable
+    public Map<String, BsonDocument> getEncryptedFieldsMap() {
+        return encryptedFieldsMap;
+    }
+
+    /**
+     * Gets whether automatic analysis of outgoing commands is set.
+     *
+     * <p>Set bypassQueryAnalysis to true to use explicit encryption on indexed fields
```

Comment:
The statement is correct. Underneath, libmongocrypt has to append the schema to the query/CRUD command so that the server can finish the indexing/query.